File: tests/python/pipeline/test-edge.py
#!/usr/bin/env python
#ckwg +5
# Copyright 2011 by Kitware, Inc. All Rights Reserved. Please refer to
# KITWARE_LICENSE.TXT for licensing information, or contact General Counsel,
# Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.


def log(msg):
    import sys
    sys.stderr.write("%s\n" % msg)


def test_import():
    try:
        import vistk.pipeline.edge
    except:
        log("Error: Failed to import the edge module")


def test_create():
    from vistk.pipeline import config
    from vistk.pipeline import edge

    c = config.empty_config()

    edge.Edge(c)
    edge.Edges()


def test_datum_create():
    from vistk.pipeline import datum
    from vistk.pipeline import edge
    from vistk.pipeline import stamp

    d = datum.complete()
    s = stamp.new_stamp()

    edge.EdgeDatum(d, s)
    edge.EdgeData()


def test_api_calls():
    from vistk.pipeline import config
    from vistk.pipeline import datum
    from vistk.pipeline import edge
    from vistk.pipeline import modules
    from vistk.pipeline import process_registry
    from vistk.pipeline import stamp

    c = config.empty_config()

    e = edge.Edge(c)

    e.makes_dependency()
    e.has_data()
    e.full_of_data()
    e.datum_count()

    d = datum.complete()
    s = stamp.new_stamp()

    ed = edge.EdgeDatum(d, s)

    e.push_datum(ed)
    e.get_datum()

    e.push_datum(ed)
    e.peek_datum()
    e.pop_datum()

    e.set_required_by_downstream(True)
    e.required_by_downstream()

    modules.load_known_modules()

    reg = process_registry.ProcessRegistry.self()

    p = reg.create_process('orphan', c)

    e.set_upstream_process(p)
    e.set_downstream_process(p)


def test_datum_api_calls():
    from vistk.pipeline import datum
    from vistk.pipeline import edge
    from vistk.pipeline import stamp

    d = datum.complete()
    s = stamp.new_stamp()

    ed = edge.EdgeDatum(d, s)

    ed.datum
    ed.datum = d
    ed.stamp
    ed.stamp = s


def main(testname):
    if testname == 'import':
        test_import()
    elif testname == 'create':
        test_create()
    elif testname == 'datum_create':
        test_datum_create()
    elif testname == 'api_calls':
        test_api_calls()
    elif testname == 'datum_api_calls':
        test_datum_api_calls()
    else:
        log("Error: No such test '%s'" % testname)


if __name__ == '__main__':
    import os
    import sys

    if not len(sys.argv) == 4:
        log("Error: Expected three arguments")
        sys.exit(1)

    testname = sys.argv[1]

    os.chdir(sys.argv[2])

    sys.path.append(sys.argv[3])

    main(testname)

File: tests/lua/pipeline/test-edge.lua
#!/usr/bin/env lua
--ckwg +5
-- Copyright 2011 by Kitware, Inc. All Rights Reserved. Please refer to
-- KITWARE_LICENSE.TXT for licensing information, or contact General Counsel,
-- Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.


function log(msg)
    io.stderr:write(string.format("%s\n", msg))
end


function test_import()
    if not pcall(require, "vistk.pipeline.edge") then
        log("Error: Failed to import the edge module")
    end
end


function test_create()
    require("vistk.pipeline.edge")
    require("vistk.pipeline.config")

    local c = vistk.pipeline.empty_config()

    vistk.pipeline.edge(c)
    vistk.pipeline.edges()
end


function test_datum_create()
    require("vistk.pipeline.datum")
    require("vistk.pipeline.edge")
    require("vistk.pipeline.stamp")

    local d = vistk.pipeline.complete_datum()
    local s = vistk.pipeline.new_stamp()

    vistk.pipeline.edge_datum(d, s)
    vistk.pipeline.edge_data()
end


function test_api_calls()
    require("vistk.pipeline.config")
    require("vistk.pipeline.datum")
    require("vistk.pipeline.edge")
    require("vistk.pipeline.modules")
    require("vistk.pipeline.process_registry")
    require("vistk.pipeline.stamp")

    local c = vistk.pipeline.empty_config()

    local e = vistk.pipeline.edge(c)

    e:makes_dependency()
    e:has_data()
    e:full_of_data()
    e:datum_count()

    local d = vistk.pipeline.complete_datum()
    local s = vistk.pipeline.new_stamp()

    local ed = vistk.pipeline.edge_datum(d, s)

    e:push_datum(ed)
    e:get_datum()

    e:push_datum(ed)
    e:peek_datum()
    e:pop_datum()

    vistk.pipeline.load_known_modules()

    local reg = vistk.pipeline.process_registry.self()

    local p = reg:create_process('orphan', c)

    e:set_upstream_process(p)
    e:set_downstream_process(p)

    e:mark_downstream_as_complete()
    e:is_downstream_complete()
end


function test_datum_api_calls()
    require("vistk.pipeline.datum")
    require("vistk.pipeline.edge")
    require("vistk.pipeline.stamp")

    local d = vistk.pipeline.complete_datum()
    local s = vistk.pipeline.new_stamp()

    local ed = vistk.pipeline.edge_datum(d, s)

    local tmp
    tmp = ed.datum
    ed.datum = d
    tmp = ed.stamp
    ed.stamp = s
end


function main(testname)
    if testname == 'import' then
        test_import()
    elseif testname == 'create' then
        test_create()
    elseif testname == 'datum_create' then
        test_datum_create()
    elseif testname == 'api_calls' then
        test_api_calls()
    elseif testname == 'datum_api_calls' then
        test_datum_api_calls()
    else
        log(string.format("Error: No such test '%s'", testname))
    end
end


if #arg ~= 2 then
    log("Error: Expected two arguments")
    os.exit(1)
end

local testname = arg[1]

package.cpath = string.format("%s/?@CMAKE_SHARED_LIBRARY_SUFFIX@;%s", arg[2], package.cpath)

local status, err = pcall(main, testname)

if not status then
    log(string.format("Error: Unexpected exception: %s", err))
end
